### ISSUE TITLE
DQN on MountainCar

### ISSUE THREAD

**Issue by qsh-zh** [vwxyzjn/cleanrl#255]
<details>
<summary>Details</summary>

</details>

## Problem Description

Pytorch DQN fails on MountainCar. Try two settings in [the issue](https://github.com/vwxyzjn/cleanrl/issues/156)

## Checklist

- [x] I have installed dependencies via `poetry install` (see [CleanRL's installation guideline](https://docs.cleanrl.dev/get-started/installation/).
- [x] I have checked that there is no similar [issue](https://github.com/vwxyzjn/cleanrl/issues) in the repo (required)

## Current Behavior

![image](https://user-images.githubusercontent.com/28699575/183264352-04270732-5b87-4fdf-ace9-025ae1ddb4ab.png)

## Expected Behavior

DQN should learn the policy.

## Possible Solution

Not sure what can be done. Quite surprising that DQN fails on the simple env.

## Steps to Reproduce

Modifications in `hotfix` are same as [the issue](https://github.com/vwxyzjn/cleanrl/issues/156)
```shell
# DQN-hotfix
rb = ReplayBuffer(
        args.buffer_size,
        envs.single_observation_space,
        envs.single_action_space,
        device,
        handle_timeout_termination=False,
    )

# DQN
rb = ReplayBuffer(
        args.buffer_size,
        envs.single_observation_space,
        envs.single_action_space,
        device,
        handle_timeout_termination=True,
    )
```

**Comment by vwxyzjn** [comment]
Hello, thanks for reporting. Could you check if your performance match the reported performance in the docs? https://docs.cleanrl.dev/rl-algorithms/dqn/#experiment-results_1

**Comment by qsh-zh** [comment]
@vwxyzjn Thanks for your fast response. I think the performance almost matches what we have in the docs. 

Except for the second random seed, seed=1/3 has a very similar behavior in my experiments.

Do you think the unsatisfying is due to suboptimal hyperparameters? Or DQN can not do well in the challenging env?

Thanks,

**Comment by vwxyzjn** [comment]
Yeah it is unsatisfactory. We always welcome new contributors! If you are interested in trying out https://github.com/vwxyzjn/cleanrl/pull/228 to find a set of params that work well for `CartPole-v1`, `MountainCar-v0`, and `Acrobot-v1`, that will be great.
